src/Compiler/Microsoft.AspNetCore.Razor.Language/src/BoundAttributeDescriptor.cs

```diff
@@ -91,7 +92,27 @@ public bool HasIndexer
         protected set => SetOrClearFlag(HasIndexerBit, value);
     }
 
-    public string Documentation { get; protected set; }
+    public string Documentation
+    {
+        get
+        {
+            return _documentationObject switch
+            {
+                string s => s,
+                DocumentationDescriptor d => d.GetText(),
+                null => null,
+                _ => throw new NotSupportedException()
+            };
+        }
+
+        protected set => _documentationObject = value;
+    }
+
+    internal object DocumentationObject
+    {
+        get => _documentationObject;
+        set => _documentationObject = value;
```

Comment:
Could we verify that this is either `null` or a `DocumentationDescriptor`?

Reply:
Sure. Will do.